src/v2/view-builder/views/idp/AuthenticatorIdPView.js

```diff
@@ -0,0 +1,47 @@
+import { loc } from 'okta';
+import BaseForm from '../../internals/BaseForm';
+import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
+import AuthenticatorEnrollFooter from '../../components/AuthenticatorEnrollFooter';
+import AuthenticatorVerifyFooter from '../../components/AuthenticatorVerifyFooter';
+
+const Body = BaseForm.extend({
+
+  title () {
+    return this.options.isChallenge
+      ? loc('oie.customFactor.challenge.title', 'login', [this.options.displayName])
+      : loc('oie.customFactor.enroll.title', 'login', [this.options.displayName]);
+  },
+
+  subtitle () {
+    return this.options.isChallenge
+      ? loc('oie.customFactor.challenge.description', 'login', [this.options.displayName])
+      : loc('oie.customFactor.enroll.description', 'login', [this.options.displayName]);
+  },
+
+  save () {
+    return this.options.isChallenge
+      ? loc('mfa.challenge.verify', 'login')
+      : loc('mfa.enroll', 'login');
+  },
+
+});
+
+export default BaseAuthenticatorView.extend({
+  initialize () {
+    BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
+
+    const currentAuthenticator = this.options.appState.get('currentAuthenticator');
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments')?.value || [];
+    const existingAuthenticator = authenticatorEnrollments
+      .filter(authenticator => authenticator.id === currentAuthenticator.id);
```

Comment:
It might be better to use `find` or `findWhere` instead of `filter`, since you only need to know if there is a one model that matches the id, right? And that will return you just one model or undefined...
as opposed to `filter` that will go through all the elements in the array and will return you another array

Reply:
Good suggestion! Will update.